rust/chains/abacus-ethereum/src/retrying.rs

```diff
@@ -1,38 +1,57 @@
 use std::{fmt::Debug, str::FromStr, time::Duration};
 
 use async_trait::async_trait;
-use ethers::providers::{JsonRpcClient, ProviderError};
+use ethers::providers::{Http, JsonRpcClient, ProviderError};
 use serde::{de::DeserializeOwned, Serialize};
 use serde_json::Value;
 use thiserror::Error;
 use tokio::time::sleep;
 use tracing::{debug, instrument, warn};
 
+use crate::HttpClientError;
+
 /// An HTTP Provider with a simple naive exponential backoff built-in
 #[derive(Debug, Clone)]
 pub struct RetryingProvider<P> {
     inner: P,
     max_requests: usize,
+    base_retry_ms: u64,
 }
 
 impl<P> RetryingProvider<P> {
     /// Instantiate a RetryingProvider
-    pub fn new(inner: P, max_requests: usize) -> Self {
-        Self {
+    pub fn new(inner: P, max_requests: usize, base_retry_ms: u64) -> Self {
+        let mut zelf = Self {
```

Comment:
optional nit, why not just set max_requests and base_retry_ms and avoid the zelf thing?

Reply:
Did it this way because I wanted to use the constraints defined in the setters without putting the constraints in two different places.